x/auth/keeper/migrations.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	v043 "github.com/cosmos/cosmos-sdk/x/auth/legacy/v043"
+	"github.com/gogo/protobuf/grpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Migrator is a struct for handling in-place store migrations.
+type Migrator struct {
+	keeper      AccountKeeper
+	queryServer grpc.Server
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(keeper AccountKeeper, queryServer grpc.Server) Migrator {
+	return Migrator{keeper: keeper, queryServer: queryServer}
+}
+
+// Migrate1to2 migrates from version 1 to 2.
+func (m Migrator) Migrate1to2(ctx sdk.Context) error {
+	wb, err := v043.MigrateStore(ctx, m.keeper.GetAllAccounts(ctx), m.queryServer)
```

Comment:
`m.keeper.GetAllAccounts(ctx)` can be painful. Instead of getting all accounts here, and then saving them in a loop in line 29, why not using an iterator and saving the account right after updating it?

Reply:
Let's validate how this impacts the runtime and how long it will take to do a migration.

Reply:
Agreed, in the end we were already using iterators before.

I might refactor the MigrateStore to accept an account instead of a slice of accounts, let it do its magic and then write back data all in the iterator loop.

WDYT?

Reply:
Yes, this is the right way :+1:

Reply:
LFTM, will implement ASAP.

Reply:
Yeah, what happened to the iterator? :D